### PR TITLE
Added extra slash to remove slash from end of URL.

### PR DESCRIPTION
**Pull request by yashcho**
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

The issue is with the JENKINS_UC_LATEST URL it is format in wrong way that makes download of plugin with error.

Example :
jenkinsUcJson=$(curl ${CURL_OPTIONS:--sSfL} -o /dev/null -w "%{url_effective}" "https://updates.jenkins.io/update-center.json?version=2.303") 
JENKINS_UC_LATEST=${jenkinsUcJson//update-center.json} gives https://updates.jenkins.io/dynamic-2.291/ which will fail to download plugins as now the url will be https://updates.jenkins.io/dynamic-2.291//latest/${plugin}.hpi"

so to overcome we can add one more slash as shown in commit and then the URL will look like https://updates.jenkins.io/dynamic-2.291/latest/${plugin}.hpi

### CODE REVIEW

**Review by MarkEWaite — COMMENTED** [review]
First preference is that people switch from using the install-plugins.sh script to use the plugin installation manager tool that is available through the `jenkins-plugin-cli.sh` script in the docker image.  It includes many more features than install-plugins.sh.  It also handles retrieving plugins from the Jenkins update center more accurately than the install-plugins.sh script does.

The `2.303` in the example is referring to the weekly release of Jenkins 2.303.  The Jenkins project has not supported the weekly release of Jenkins 2.303 for over 16 weeks.  The choice of that version seems to be due to the `jenkinsMajorMinorVersion` function in the `install-plugins.sh` file.  It drops the final component of the LTS version number and causes the update center lookup to return a different URL than would be returned if the LTS version 2.303.3 had been provided to the curl command.

I'd prefer to not make any changes to this shell script.

**Comment by yashcho** [comment]
Yes I agree to your point of using jenkins-plugins-cli but as an improvement for install-plugins.sh this can changed still, as you said the LTS is 2.303.3 so when I provide it to curl it returns "https://updates.jenkins.io/dynamic-stable-2.303.3/update-center.json" which is write URL but as if got to next step of trimming the /update-center.json part.

JENKINS_UC_LATEST=${jenkinsUcJson//update-center.json/} this will return "https://updates.jenkins.io/dynamic-stable-2.303.3/" which will form the url for line number 89 as "https://updates.jenkins.io/dynamic-stable-2.303.3//latest/${plugins}.hpi" which will obviously return 404 due to extra slash "/".

So my change is just trimming the extra slash from end to make the download url at line 89 look like "https://updates.jenkins.io/dynamic-stable-2.303.3/latest/${plugins}.hpi"

In short my suggestion is to improve the download URL so as it is able to download the plugins.. :)

**Comment by MarkEWaite** [comment]
> So my change is just trimming the extra slash from end to make the download url at line 89 look like "[https://updates.jenkins.io/dynamic-stable-2.303.3/latest/${plugins}.hpi](https://updates.jenkins.io/dynamic-stable-2.303.3/latest/$%7Bplugins%7D.hpi)"
> 
> In short my suggestion is to improve the download URL so as it is able to download the plugins.. :)

The Jenkins update center downloads from all the following URLs (many that include multiple adjacent '/' characters)

* https://updates.jenkins.io/dynamic-stable-2.303.3/latest/elastic-axis.hpi
* https://updates.jenkins.io/dynamic-stable-2.303.3/latest//elastic-axis.hpi
* https://updates.jenkins.io/dynamic-stable-2.303.3///latest/////elastic-axis.hpi
* https://updates.jenkins.io/dynamic-stable-2.303.3///latest////elastic-axis.hpi
* https://updates.jenkins.io/dynamic-2.291/latest/elastic-axis.hpi
* https://updates.jenkins.io/dynamic-2.291/latest//elastic-axis.hpi
* https://updates.jenkins.io/dynamic-2.291///latest/////elastic-axis.hpi
* https://updates.jenkins.io/dynamic-2.291///latest////elastic-axis.hpi

I don't see the benefit of the change when all the cases that I check are returning the expected download of the file.

Can you explain further why you believe that the download URL with extra '/' characters will report 404?

**Comment by yashcho** [comment]
Sorry my bad, when I was upgrading Jenkins to 2.190.3 I saw that extra "/' was giving 404 while installing it through install-plugin.sh file but now I tried once again and it is working without removing extra "/".

Was there any change regarding the redirection between those version? -> If no, then please close my PR.. sorry for that.. :)

**Comment by MarkEWaite** [comment]
Thanks for the clarification!  I've closed the pull request.

The Jenkins update center only supports releases within the last year.  The [web page](https://updates.jenkins.io/) says:

> At the moment, Jenkins weekly and LTS releases up to a year old are supported; anything older will receive update metadata for the oldest supported releases.

I believe that means the preceding call to request the update center for 2.190.3 may have returned an empty string or an incorrect URL because 2.190.3 is more than a year old.
